### PR TITLE
Added reason to the WebSocket close

### PR DESCRIPTION
**Pull request by pgjones**
This is allowed as per RFC 6455 but previously not supported in
ASGI. This allows servers/applications to use the reason if desired.

See also #234

### CODE REVIEW

File: specs/www.rst [hunk 1]
```diff
@@ -413,6 +414,9 @@ with the close code passed in the message (or 1000 if none is specified).
 * ``code`` (*int*) -- The WebSocket close code, as per the WebSocket
   spec.  Optional; if missing defaults to ``1000``.
 
+* ``reason`` (*Unicode string*) -- A reason given for the closure, can
+  be any string. Optional; if missing default is empty string.
```
Comment (andrewgodwin) [inline specs/www.rst:418]: Given the type declared, should this be "if missing or None"?

Comment (pgjones) [reply specs/www.rst:418]: I've gone for the same wording as the [`query_string`](https://github.com/django/asgiref/blame/master/specs/www.rst#L94), should I change both?

Comment (andrewgodwin) [reply specs/www.rst:418]: Yes, let's update both, explicit is better here I think.

Comment (pgjones) [reply specs/www.rst]: Done.

**Comment by pgjones** [comment]
(Sorry messed up the branch, now ready)

**Comment by andrewgodwin** [comment]
Thanks for getting this done!

**Comment by Kludex** [comment]
Shouldn't I be able to see 2.3 instead of 2.1 on [this page](https://asgi.readthedocs.io/en/latest/specs/www.html) by now? Am I missing something here?

**Comment by andrewgodwin** [comment]
My bad - looks like type annotations being added killed the build as it was still on Python 2, somehow. Should be all fixed now (we're up to version 3)

**Comment by Kludex** [comment]
Perfect! Thanks @andrewgodwin ! :)

**Pull request merged**